apollo-ios/Design/swift-identifiable.md

```diff
@@ -1,29 +1,47 @@
 # Summary
 
-This document proposes a new GraphQL directive named `@apollo_client_ios_identity`, to be used to mark a field which can uniquely identify an object.
+This document proposes a new GraphQL directive named `@identity`, to be used to mark a field which can uniquely identify an object.
 
 # Introduction
 
 SwiftUI makes heavy use of the [Identifiable](https://developer.apple.com/documentation/swift/identifiable) protocol, which is used to track the identity of entities across data changes. If an object with a List is replaced with a different object with the same identity, SwiftUI will animate the item changing instead of animating an insertion. Objects that do not conform to the Identifiable protocol require additional boilerplate to be usable inside SwiftUI.
 
 Apollo Client iOS could assist the developer by adding conformance to the Identifiable protocol to its generated models. Selecting the field to be used as an identity is done by adding a new GraphQL directive.
 
+A concept of identity is required to allow response objects to be cached. The various Apollo Client projects have mechanisms that allow for identifiers to be selected through additional code. The new directive could allow schema authors to assist client authors with caching.
+
 # Definition
 
 The directive is defined as:
 ```graphql
-directive @apollo_client_ios_identity on FIELD | FIELD_DEFINITION
+directive @identity(scope: IdentityScope = SELECTION) on FIELD | FIELD_DEFINITION
+
+enum IdentityScope {
```

Comment:
This is a novel idea but you can probably derive the scope from the context in which the directive is used; interface field vs. type field vs. selection field.

This may also introduce a bunch of validation required to make sure it isn't used with excessive scope, i.e.: should a selection field be able to define global scope? _Probably not._

Reply:
I changed my mind regarding the SELECTION scope. Its application is too limited (protocol conformance only), so I took it out. I think removing this case also removes the reason for clients to widen the scope themselves.